RecognizeDtmf/RecognizeDtmf.cs

```diff
@@ -0,0 +1,341 @@
+                    }
+
+                    await HangupAsync().ConfigureAwait(false);
+                }
+
+                // Wait for the call to terminate
+                await callTerminatedTask.Task.ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, $"Call ended unexpectedly, reason: {ex.Message}");
+            }
+        }
+
+        private async Task<CallConnection> CreateCallAsync(string targetPhoneNumber)
+        {
+            try
+            {
+                //Preparting request data
+                CallSource source = new CallSource(new CommunicationUserIdentifier(callConfiguration.SourceIdentity));
+                source.CallerId = new PhoneNumberIdentifier(callConfiguration.SourcePhoneNumber);
+                var target = new PhoneNumberIdentifier(targetPhoneNumber);
+
+                var createCallOption = new CreateCallOptions(source,
+                    new List<CommunicationIdentifier>() { target },
+                    new Uri(callConfiguration.AppCallbackUrl));
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Performing CreateCall operation");
+
+                var call = await callClient.CreateCallAsync(createCallOption).ConfigureAwait(false);
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"CreateCallConnectionAsync response --> {call.GetRawResponse()}, Call Connection Id: { call.Value.CallConnectionProperties.CallConnectionId}");
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"Call initiated with Call Connection id: { call.Value.CallConnectionProperties.CallConnectionId}");
+
+                RegisterToCallStateChangeEvent(call.Value.CallConnectionProperties.CallConnectionId);
+
+                //Wait for operation to complete
+                await callEstablishedTask.Task.ConfigureAwait(false);
+
+                return call.Value.CallConnection;
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, string.Format("Failure occured while creating/establishing the call. Exception: {0}", ex.Message));
+                throw ex;
+            }
+        }
+
+        private async Task StartRecognizingDtmf(string targetPhoneNumber)
+        {
+            if (reportCancellationToken.IsCancellationRequested)
+            {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Cancellation request, PlayAudio will not be performed");
+                return;
+            }
+
+            try
+            {
+                string audioFilePath = callConfiguration.AudioFileUrl + callConfiguration.AudioFileName;
+                PlaySource audioFileUri = new FileSource(new Uri(audioFilePath));
+
+                // listen to play audio events
+                RegisterToPlayAudioResultEvent(callConnection.CallConnectionId);
```

Comment:
As we discussed before, This should be RegisterToRecognizeResultEvent, once recognize api is initiated we expected back `RecognizeCompleted` event and not play event

Reply:
We are not awaiting for `RecoginzeCompleted` or `RecoginzeFailed` event here, once we get back `RecognizeCompleted` (in case customer enters DTMF) or `RecognizeFailed` event (in case where customer do not enter any dtmf). once we get back either of the event we should hangup the call.